### PR TITLE
feat(nuxt): add `page:view-transition:start` hook

### LINKED ISSUE #26044: Provide a way to obtain the ongoing ViewTransition object
### Describe the feature

The current implementation of the experimental ViewTransition plugin is great, but it can be limiting not having the actual transition object returned by 'document.startViewTransition'. This object can be used to wait for certain events during the transition (eg.: finished).

It would be great to receive this object from nuxt when one is created.
This would be needed in Elks (in-progress) view transition implementation as described in this issue here: https://github.com/elk-zone/elk/pull/2211#issuecomment-1974770960

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by horvbalint**
### 🔗 Linked issue

https://github.com/nuxt/nuxt/issues/26044

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This is a small PR that introduces a new client lifecycle hook, which is called right after a [ViewTransition object](https://developer.mozilla.org/en-US/docs/Web/API/ViewTransition) is created by the [experimental view transition plugin](https://nuxt.com/docs/getting-started/transitions#view-transitions-api-experimental).

Using this hook an app can obtain the transition object.

Resolves #26044

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have added tests (if possible).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by horvbalint** [comment]
I am not sure, how to solve the not exported interface error in a way so that it can be imported in projects using nuxt.

File: packages/nuxt/src/app/nuxt.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ export interface RuntimeNuxtHooks {
   'page:finish': (Component?: VNode) => HookResult
   'page:transition:start': () => HookResult
   'page:transition:finish': (Component?: VNode) => HookResult
+  'page:view-transition:start': (transition: ViewTransition) => HookResult
```
Comment (GalacticHypernova) [inline packages/nuxt/src/app/nuxt.ts:48]: How about providing an exported type in nuxt and importing it here?
```ts
export interface ViewTransition {
    ready: Promise<void>
    finished: Promise<void>
    updateCallbackDone: Promise<void>
  }
  ```
  
  ```ts
import {ViewTransition} from "wherever you'll place it"
```

Comment (horvbalint) [reply packages/nuxt/src/app/nuxt.ts:49]: Yeah I think I overthinked this one 😅 I tought it has to be importable as "import type {ViewTransition} from 'nuxt'"

Comment (GalacticHypernova) [reply packages/nuxt/src/app/nuxt.ts:49]: Happens to the best of us =)

File: packages/nuxt/src/app/plugins/view-transitions.client.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ export default defineNuxtPlugin((nuxtApp) => {
       changeRoute()
       return promise
     })
+    nuxtApp.callHook('page:view-transition:start', transition)
```
Comment (huang-julien) [inline packages/nuxt/src/app/plugins/view-transitions.client.ts:38]:
```suggestion
    await nuxtApp.callHook('page:view-transition:start', transition)
```
We probably need to await the hook

File: docs/3.api/6.advanced/1.hooks.md [hunk 3]
```diff
@@ -28,6 +28,7 @@ Hook                   | Arguments           | Environment     | Description
 `page:loading:start`   | -                   | Client          | Called when the `setup()` of the new page is running.
 `page:loading:end`     | -                   | Client          | Called after `page:finish`
 `page:transition:finish`| `pageComponent?`    | Client          | After page transition [onAfterLeave](https://vuejs.org/guide/built-ins/transition.html#javascript-hooks) event.
+`page:view-transition:start` | `transition`        | Client          | Called after document.startViewTransition is called by the [experimental nuxt plugin](https://nuxt.com/docs/getting-started/transitions#view-transitions-api-experimental).
```
Comment (danielroe) [inline docs/3.api/6.advanced/1.hooks.md:31]:
```suggestion
`page:view-transition:start` | `transition`        | Client          | Called after `document.startViewTransition` is called when [experimental viewTransition support is enabled](https://nuxt.com/docs/getting-started/transitions#view-transitions-api-experimental).
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**